grpcService/pkg/utils/queries.go

```diff
@@ -1,8 +1,41 @@
 package utils
 
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/timoteoBone/microservice-project/grpcService/pkg/entities"
+)
+
 var (
 	CreateUserQuery  string = "INSERT INTO USER (first_name, id, pass, age, email) VALUES (?,?,?,?,?)"
 	GetUserQuery     string = "SELECT first_name, age, email FROM USER WHERE id=?"
 	GetPasswordQuery string = "SELECT pass FROM USER WHERE id = ?"
 	DeleteUserQuery  string = "DELETE FROM USER WHERE id = ?"
+	UpdateUserQuery  string = "UPDATE USER SET first_name=?, pass=?, age=?, email=? WHERE id = ?"
 )
+
+func GenerateQuery(user entities.User) (string, []interface{}) {
+	referencesToSQL := entities.FieldsReferenceSql
+
+	var query strings.Builder
+	query.WriteString("UPDATE USER SET")
+
+	var args []interface{}
+
+	v := reflect.ValueOf(user)
```

Comment:
refection is a great tool to use, nevertheless it impacts the performance, and you are coupling the code to the columns names